src/main/java/com/urswolfer/gerrit/client/rest/http/changes/ChangeApiRestClient.java

```diff
@@ -63,6 +67,8 @@
     private final EditInfoParser editInfoParser;
     private final CommitInfoParser commitInfoParser;
     private final String id;
+    private final ServerRestClient serverRestClient;
+    private final AtomicReference<String> version = new AtomicReference<String>();
```

Comment:
This will not behave nicely when we change the Gerrit service at runtime. And it is also not reusable by other parts of this lib which probably also could use the server version.

I've not looked into it in detail, but it would might be an idea to put the version into `LoginCache` (because that cache gets invalidated when we connect to another server or when a session expires), see https://github.com/uwolfer/gerrit-rest-java-client/blob/master/src/main/java/com/urswolfer/gerrit/client/rest/http/GerritRestClient.java#L98

Reply:
Let me explore your suggestion. If the moving parts gets too big, I'll address the point in a separate PR. WDYT?

Reply:
> This will not behave nicely when we change the Gerrit service at runtime. 

You cannot change the Gerrit URL dynamically on a `ChangeApiRestClient` AFAIK.
If you change the endpoint, you will have a new instance of the `ChangeApiRestClient` and thus another version string.

> And it is also not reusable by other parts of this lib which probably also could use the server version.

That is a valid concern. However, it really depends on why you want to get the version.
API caching is useful but could be dangerous as well. You could actually use the version API as a `keepalive` or `watchdog` mechanism: having that result cached wouldn't help in that case.

However, you are right: having this cached version reusable is a good idea. Will look into that.

Reply:
Done and exposed as a separate method that tells that the value returned is a cached one.